From 48d00e4f6ff7bcb2aa400a09e08bf98f23277530 Mon Sep 17 00:00:00 2001
From: Jin Xue <jin.xue@intel.com>
Date: Tue, 27 Nov 2018 16:01:12 +0800
Subject: [PATCH 3/6] output audio to HDMI displays

There are 3 alsa plugins wrapped HDMI displays. The device numbers are
3,7,8. When there are 3 HDMI cables connedted, all the alsa devices will
be available. But the number are hard-coded until now.

Change-Id: Iab820e52c38383fae4286ec1b3a1dee66afd64df
Tracked-On:
Signed-off-by: Jin Xue <jin.xue@intel.com>
Reviewed-on: 653418
Reviewed-by: Zhang, Hongyu <hongyu.zhang@intel.com>
Tested-by: Zhang, Hongyu <hongyu.zhang@intel.com>
Reviewed-on: 657227
---
 audio_hw.c | 13 ++++++++++++-
 1 file changed, 12 insertions(+), 1 deletion(-)

diff --git a/audio_hw.c b/audio_hw.c
index 04846fa..1178710 100644
--- a/audio_hw.c
+++ b/audio_hw.c
@@ -52,6 +52,11 @@
 #define IN_PERIOD_COUNT 2
 #define IN_SAMPLING_RATE 48000
 
+#define CONNECTED_DISPLAYS 3
+#define DISPLAY_SHIFT 0
+
+unsigned int hdmi_output[CONNECTED_DISPLAYS] = {3, 7, 8};
+
 struct pcm_config pcm_config_out = {
     .channels = 2,
     .rate = OUT_SAMPLING_RATE,
@@ -187,7 +192,13 @@ static int start_output_stream(struct stream_out *out)
         return -ENODEV;
     }
 
-    out->pcm = pcm_open(PCM_CARD, PCM_DEVICE, PCM_OUT | PCM_NORESTART | PCM_MONOTONIC, out->pcm_config);
+    char prop_id[PROPERTY_VALUE_MAX];
+    memset(prop_id, 0, sizeof(prop_id));
+    property_get("sys.container.id", prop_id, "0");
+    unsigned int id = atoi(prop_id) % CONNECTED_DISPLAYS;
+    unsigned int device = hdmi_output[(id + DISPLAY_SHIFT) % CONNECTED_DISPLAYS];
+
+    out->pcm = pcm_open(PCM_CARD, device, PCM_OUT | PCM_NORESTART | PCM_MONOTONIC, out->pcm_config);
     if (!out->pcm) {
 	ALOGE("pcm_open(out) failed: device not found");
         return -ENODEV;
-- 
2.7.4

